### PR TITLE
ZIO Streams documentation

### PR DESCRIPTION
**Pull request by khajavi**
- [X] Introduction
- [X] ZStream
- [X] Stream
- [x] ZSink
- [X] Sink
- [x] Transducer
- [x] Type Aliases

### CODE REVIEW

File: docs/datatypes/stream/zstream.md [hunk 1]
```diff
@@ -0,0 +1,1865 @@
...
+```scala mdoc:silent:nest
+val schedule: Schedule[Clock with Random, Chunk[Chunk[Record]], Long] =
+  // Start off with 30-second timeouts as long as the batch size is < 1000
+  Schedule.fixed(30.seconds).whileInput[Chunk[Chunk[Record]]](_.flatten.length < 100) andThen
+    // and then, switch to a shorter jittered schedule for as long as batches remain over 1000
+    Schedule.fixed(5.seconds).jittered.whileInput[Chunk[Chunk[Record]]](_.flatten.length >= 1000)
+    
+dataStream
+  .aggregateAsyncWithin(ZTransducer.collectAllN(2000), schedule)
+```
```
Comment (khajavi) [inline docs/datatypes/stream/zstream.md:1660]: Hello @iravid 
Yesterday all the time, I was working on the adaptive batching pattern which you have presented in one of your talk about ZIO streams.
I tried to rewrite and test that and and then include this nice use-case in ZIO stream documentation. But I'm not sure that this example works as what you presented.
When I debug that program, I found that in the `_.flatten.length < 100` predicate, the value of `_.flatten.length` does not always represent the batch size. Most of the time, its value is zero and sometime it is `2000`. I didn't find a situation when the size of the batch is is something between 0 and 2000. (I also jittered the upstream to make sure I have different throughout)
Could you please look at that and check if I'm right or not?

**Comment by khajavi** [comment]
Hi @jdegoes 
It is ready to be reviewed.

File: docs/datatypes/stream/zstream.md (hunk 1 above)
Comment (iravid) [reply docs/datatypes/stream/zstream.md:1690]: Hey! Will try to take a look later today

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
@khajavi Thank you! It's a huge PR and the quality of the material looks good. I think we can do followup PRs to address improvements and fixes.

**Pull request merged**